### PR TITLE
DeviceSpecification proto updates

### PR DESCRIPTION
**Pull request by verult**
Part of https://github.com/quantumlib/Cirq/issues/5050

Since each gate has its own message now, target specification could potentially be moved inside gate messages (e.g. for 2-qubit gates) or disappear entirely and embed the target set in Cirq (e.g. measurement). This is lower priority since there are no major problems with the existing setup, so deferring this for now and get back to it before Cirq 1.0 if there's time.

Micro-optimization: Leaving `GateSpecification` field numbers < 16 empty to allow for potential future common fields.

@dstrain115 @maffoo

### CODE REVIEW

File: cirq-google/cirq_google/api/v2/device.proto [hunk 1]
```diff
@@ -30,12 +33,55 @@ message DeviceSpecification {
   string developer_recommendations = 4;
 }
 
+// Replaces `GateDefinition`.
```
Comment (dstrain115) [inline cirq-google/cirq_google/api/v2/device.proto:36]: Maybe also explain the purpose of this message in addition to explaining what it is replacing (which will become increasingly obsolete over time)

File: cirq-google/cirq_google/api/v2/device.proto [hunk 2]
```diff
@@ -30,12 +33,55 @@ message DeviceSpecification {
...
+  // Valid targets that this gate can use.
+  // Values in this list correspond to the name of the TargetSet
+  // If unset, all combinations of `DeviceSpecification.valid_qubits` are
+  // allowed.
+  repeated string valid_targets = 2;
+
+  // This specifies the gate type and gate parameter constraints for this
+  // GateSpecification.
+  oneof gate {
+    Sycamore syc = 16;
```
Comment (dstrain115) [inline cirq-google/cirq_google/api/v2/device.proto:51]: Will this micro optimization really save that much space?  Maybe one byte per gate supported, and we will support at most ~10 gates?  I am not sure the confusion is worth it.

In any case, a comment should be somewhere explained that the field numbers are reserved.

Comment (verult) [reply cirq-google/cirq_google/api/v2/device.proto:51]: Sounds good, will make it sequential

**Comment by verult** [comment]
Addressed Doug's comments. Also renamed to `(Virtual|Physical)ZPow`.

**Review by dstrain115 — APPROVED** [review]
LGTM, no further comments.

**Pull request merged**